spec.html

```diff
@@ -31957,7 +31957,7 @@ <h1>Notation</h1>
             A <em>CharSet</em> is a mathematical set of characters. &ldquo;All characters&rdquo; means the CharSet containing all code point values when the _Unicode_ flag is *true*, and otherwise means the CharSet containing all code unit values.
           </li>
           <li>
-            A <em>State</em> is an ordered pair (_endIndex_, _captures_) where _endIndex_ is an integer and _captures_ is a List of _NcapturingParens_ values. States are used to represent partial match states in the regular expression matching algorithms. The _endIndex_ is one plus the index of the last input character matched so far by the pattern, while _captures_ holds the results of capturing parentheses. The _n_<sup>th</sup> element of _captures_ is either a List of characters that represents the value obtained by the _n_<sup>th</sup> set of capturing parentheses or *undefined* if the _n_<sup>th</sup> set of capturing parentheses hasn't been reached yet. Due to backtracking, many States may be in use at any time during the matching process.
+            A <em>State</em> is an ordered pair (_endIndex_, _captures_) where _endIndex_ is an integer and _captures_ is a List of length _NcapturingParens_, indexed 1 through _NcapturingParens_. States are used to represent partial match states in the regular expression matching algorithms. The _endIndex_ is one plus the index of the last input character matched so far by the pattern, while _captures_ holds the results of capturing parentheses. The _n_<sup>th</sup> element of _captures_ is either a List of characters that represents the value obtained by the _n_<sup>th</sup> set of capturing parentheses or *undefined* if the _n_<sup>th</sup> set of capturing parentheses hasn't been reached yet. Due to backtracking, many States may be in use at any time during the matching process.
```

Comment:
I don't like this 1-indexed list weirdness. I prefer the approach I took in https://github.com/tc39/ecma262/pull/2142#discussion_r472550811.

Reply:
Conversation moved to https://github.com/tc39/ecma262/pull/2150.